### PR TITLE
Filter design docs

### PR DESCRIPTION
**Pull request by mayya-sharipova**
Filter design docs after there were received from Cloudant:
1. Added function FilterDDocs.filter in FilterUtil,
   so that a filter can be applied in JsonStoreDataAccess
2. Modified function JsonStoreDataAccess.getOne()
 to retrieve one NOT ddoc document.
 Because usual (not design docs) can have an _id that is bigger

> _design and also ids that is smaller than _design, had to use
>  two functions to retrieve non design documents:
>  getOneUrlExcludeDDoc1() and getOneUrlExcludeDDoc2()

BugzID: 56450

### CODE REVIEW

File: cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/JsonStoreDataAccess.scala [hunk 1]
```diff
@@ -48,7 +48,11 @@ class JsonStoreDataAccess (config: JsonStoreConfig)  {
   }
 
   def getOne()( implicit columns: Array[String] = null) = {
-    this.getQueryResult[Seq[String]](config.getOneUrl, processAll)
+    var r = this.getQueryResult[Seq[String]](config.getOneUrlExcludeDDoc1(), processAll)
+    if (r.size == 0 ){
+      r = this.getQueryResult[Seq[String]](config.getOneUrlExcludeDDoc2(), processAll)
+    }
```
Comment (HolgerKache) [inline cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/JsonStoreDataAccess.scala:55]: What if we have no result for the 2nd query either?

File: cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/JsonStoreConfig.scala [hunk 2]
```diff
@@ -37,6 +37,8 @@ trait JsonStoreConfig {
   implicit val bulkSize: Int 
   def allowPartition(): Boolean = {true}
   def getOneUrl(): String
+  def getOneUrlExcludeDDoc1(): String = {""}
+  def getOneUrlExcludeDDoc2(): String = {""}
```
Comment (HolgerKache) [inline cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/JsonStoreConfig.scala]: Let's not initialize these 2 so we can use proper null checking anywhere else in the code.

File: cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/JsonStoreDataAccess.scala (hunk 1 above)
Comment (mayya-sharipova) [inline cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/JsonStoreDataAccess.scala:55]: Added to raise an exception if there is no result for the 2nd query

**Comment by HolgerKache** [comment]
+1

**Pull request merged**